department-of-veterans-affairs/caseflow
# Edit request issues on claims

Connects #6212 

### Description 🚧 
Adds and hooks up a button for saving selected rated issues on a higher level claim or supplemental claim.

Implements:
* SaveButton and SelectIssuesButtons components for editing rated issues for an intake
* Associated redux actions / reducers
* PATCH /higher_level_reviews/:claim_id endpoint for updating rated issues
* PATCH /supplemental_claims/:claim_id endpoint for updating rated issues

This demo shows all the error states, but the 'no change' should not be possible to reach in the UI as the button is disabled until changes are made.
![demo](https://user-images.githubusercontent.com/279406/45711163-885d1180-bb3d-11e8-86eb-daf0c95b4145.gif)

### Acceptance Criteria
- [x] Feature tests
- [x] Clicking the "Save" button should trigger changing the issues.
- [x] Error messaging
- [x] Success messaging? (could come in a followup)
- [x] The "Save" button is grayed out until a change is made, then it's the default blue button color.
- [x] When the issues are saved, if issues of the same type (rated or non-rated) are added or removed to/from an existing EP, the new contentions are saved in VBMS under the same EP code and label (done via prior pr https://github.com/department-of-veterans-affairs/caseflow/pull/6781)

### Testing Plan
1. Select BVAAABSHIRE user (Judge with hearings and cases)
1. Pick a vbms_id out of bgs_setup.csv with has_rating (e.g. 231439628)
1. Complete intake of a request for a higher level review with the selected vbms_id
1. Look up the claim_id of the new EP
1. Visit /higher_level_reviews/:claim_id/edit
1. Select/deselect issues and save to test the following cases
   1. No change -> button is disabled gray
   1. No issues selected -> hitting save displays a 'no issues selected' error message
   1. Some or all issues selected -> hitting save displays a 'success' message
   1. While saving, the button should be disabled
